File: cpp/build-company/Tool.cpp
#include "Tool.hpp"
#include <iostream>

// Реализация конструктора
Tool::Tool(const std::string& name, int count, ToolType type)
    : Equipment(name, count), type(type) {
}

// Реализация деструктора
Tool::~Tool() {
    std::cout << "Вызван деструктор Tool для: " << name << std::endl;
}

// Реализация геттера
ToolType Tool::getType() const {
    return type;
}

// Реализация сеттера
void Tool::setType(ToolType type) {
    this->type = type;
}

// Реализация функции отображения информации
void Tool::displayInfo() const {
    std::cout << "=== Инструмент ===" << std::endl;
    std::cout << "Название: " << name << std::endl;
    std::cout << "Тип: " << toolTypeToString(type) << std::endl;
    std::cout << "Количество: " << count << " ед." << std::endl;
}

File: csharp/build-company/Tool.cs
namespace BuildCompany;

/// <summary>
/// Класс для инструментов и ручного оборудования
/// </summary>
public class Tool : Equipment
{
    private ToolType type;

    /// <summary>
    /// Конструктор
    /// </summary>
    /// <param name="name">Название инструмента</param>
    /// <param name="count">Количество</param>
    /// <param name="type">Тип инструмента</param>
    public Tool(string name, int count, ToolType type)
        : base(name, count)
    {
        this.type = type;
    }

    /// <summary>
    /// Получить тип инструмента
    /// </summary>
    public ToolType GetToolType()
    {
        return type;
    }

    /// <summary>
    /// Установить тип инструмента
    /// </summary>
    public void SetToolType(ToolType type)
    {
        this.type = type;
    }

    /// <summary>
    /// Отображение информации об инструменте
    /// </summary>
    public override void DisplayInfo()
    {
        Console.WriteLine($"[Tool]");
        Console.WriteLine($"  Name: {name}");
        Console.WriteLine($"  Type: {type.ToDisplayString()}");
        Console.WriteLine($"  Count: {count}");
    }
}
